Source/engine/assets.cpp

```diff
@@ -473,4 +464,29 @@ void LoadHellfireArchives()
 		DisplayFatalErrorAndExit(_("Some Hellfire MPQs are missing"), _("Not all Hellfire MPQs were found.\nPlease copy all the hf*.mpq files."));
 }
 
+void UnloadModArchives()
+{
+#ifndef UNPACKED_MPQS
+	for (auto it = MpqArchives.begin(); it != MpqArchives.end();) {
+		if ((it->first >= 8000 && it->first < 9000) || it->first >= 10000) {
+			it = MpqArchives.erase(it); // erase returns the next valid iterator
+		} else {
+			++it;
+		}
+	}
+#endif
+}
+
+void LoadModArchives(std::vector<std::string_view> modnames)

```

Comment:
The argument here should have the type `std::span<const std::string_view> modname`.

A vector vs span relationship is similar to string vs string_view.